components/dashboard/src/data/featureflag-query.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { Organization, Project, User } from "@gitpod/gitpod-protocol";
+import { useQuery } from "@tanstack/react-query";
+import { getExperimentsClient } from "../experiments/client";
+import { useCurrentProject } from "../projects/project-context";
+import { useCurrentUser } from "../user-context";
+import { useCurrentOrg } from "./organizations/orgs-query";
+
+const featureFlags = {
+    start_with_options: false,
+    showUseLastSuccessfulPrebuild: false,
+    usePublicApiWorkspacesService: false,
+    enablePersonalAccessTokens: false,
+    oidcServiceEnabled: false,
+    // Default to true to enable on gitpod dedicated until ff support is added for dedicated
+    orgGitAuthProviders: true,
+    userGitAuthProviders: false,
+    newSignupFlow: false,
+    linkedinConnectionForOnboarding: false,
+    paymentVerificationFlow: false,
+    team_only_attribution: false,
+};
+
+export const useFeatureFlag = (featureFlag: keyof typeof featureFlags) => {
+    const user = useCurrentUser();
+    const org = useCurrentOrg().data;
+    const project = useCurrentProject().project;
+
+    return useQuery<boolean>({
+        queryKey: queryKey(featureFlag, user, org, project),
```

Comment:
`gitpodHost` would be nice to have in there, too ([source](https://github.com/gitpod-io/gitpod/blob/ee82d153e3ce76e13c3987d1567ad6c1cc53df27/components/gitpod-protocol/src/experiments/types.ts#L30)).